prime-router/src/main/kotlin/cli/main.kt

```diff
@@ -256,14 +256,22 @@ class ProcessData : CliktCommand(
                 val outputFile = if (outputFileName != null) {
                     File(outputFileName!!)
                 } else {
-
+                    // is this config HL7?
+                    val hl7Config = report.destination?.translation as? Hl7Configuration?
+                    // if it is, get the test processing mode
+                    val processingMode = if (hl7Config?.useTestProcessingMode == false) {
+                        "P"
+                    } else {
+                        "T"
+                    }
                     val fileName = Report.formFilename(
                         report.id,
                         report.schema.baseName,
                         format,
                         report.createdDateTime,
                         getNameFormat(Report.NameFormat.STANDARD),
-                        receivingOrganization ?: report.destination?.translation?.receivingOrganization
+                        receivingOrganization ?: report.destination?.translation?.receivingOrganization,
+                        processingMode
                     )
```

Comment:
I guess. I don't understand why this code doesn't use the FileUtilities function here, instead of repeating it. Or the new Report.formFileName function.